addons/misra.py

```diff
@@ -1353,6 +1364,42 @@ def misra_7_3(self, rawTokens):
             if compiled.match(tok.str):
                 self.reportError(tok, 7, 3)
 
+    def misra_7_4(self, data):
+        # A string literal shall not be assigned to an object unless the object's type
+        # is constant.
+        def reportErrorIfVariableIsNotConst(variable, stringLiteral):
+            if variable.valueType:
+                if variable.valueType.constness == 0:
+                    self.reportError(stringLiteral, 7, 4)
+
+        for token in data.tokenlist:
+            # Check normal variable assignment
+            if token.isString:
+                variable = getAssignedVariableToken(token)
+                if variable:
+                    reportErrorIfVariableIsNotConst(variable, token)
+
+            # Check use as function parameter
+            if isFunctionCall(token):
+                parametersUsed = getArguments(token)
+
+                functionDeclaration = None
```

Comment:
I have the feeling you can use something like `functionDeclaration = token.astOperand1.function`